behavior/MouseTracking/get_mouse_XY_pos.m

```diff
@@ -112,14 +126,38 @@
             end
 
             % Save centroids data and update plot
+            true_frame_idx = frame_chunks(idx,1)+frame_idx-1;           
+                
             if isempty(area_vector) %sometimes blob disappears, go to previous centroid
-                centroids(frame_chunks(idx,1)+frame_idx-1,:) = c_old;
-                centroid_color = 'r';
+                
+                % trial_aware mode
+                % if image is the first image of a trial, use next centroid
+                if trial_aware && find(true_frame_idx==trial_indices(:,1),1)
+                    reassign_prev_centroid = 1;
+                    lost_centroids = lost_centroids+1;
+                    centroid_color = 'm';
+                elseif trial_aware && reassign_prev_centroid == 1
```

Comment:
added this in for cases when mouse is still not found in trials following first frame